### ISSUE TITLE
Unnecessary unused_variables lint

### ISSUE THREAD

**Issue by johnthagen** [Manishearth/rust-clippy#1552]
It would be nice if `clippy` could catch an unnecessary use of `#[allow(unused_variables)]`:

```rust
#[allow(unused_variables)]
fn fun(x: u32) -> u32{
    x
}
```

Often times `#[allow(unused_variables)]` is added during development and later should be removed after all parameters are in use so that it doesn't mask a real warning in the future.

**Comment by clarcharr** [comment]
@johnthagen perhaps this could be something to ask for upstream? there already is an `unused_attribute` link in rustc iirc and this could be a good addition to it

**Comment by mcarton** [comment]
This would probably need to be upstream. Clippy does not handle `allow/warn/deny` itself.

**Comment by johnthagen** [comment]
rustc issue opened: https://github.com/rust-lang/rust/issues/39958
